From 69c08d62219215add3cbbf0716abcb835e9cd044 Mon Sep 17 00:00:00 2001
From: Shaofeng Tang <shaofeng.tang@intel.com>
Date: Thu, 27 Aug 2020 19:44:00 +0800
Subject: [PATCH] drm/i915/gt: Schedule request retirement for gen10+

In CML, "Schedule request retirement when timeline idles" might
introduce random drm init failure during reboot. This change is
to enable schedule request retirement on Gen10+

Tracked-On: OAM-91972
Signed-off-by: Romli, Khairul Anuar <khairul.anuar.romli@intel.com>
Signed-off-by: Junxiao Chang <junxiao.chang@intel.com>
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>
---
 drivers/gpu/drm/i915/gt/intel_lrc.c | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/gt/intel_lrc.c b/drivers/gpu/drm/i915/gt/intel_lrc.c
index c8449f718156..70dadbde87c9 100644
--- a/drivers/gpu/drm/i915/gt/intel_lrc.c
+++ b/drivers/gpu/drm/i915/gt/intel_lrc.c
@@ -1435,13 +1435,15 @@ __execlists_schedule_out(struct i915_request *rq,
 	 * refrain from doing non-trivial work here.
 	 */
 
-	/*
-	 * If we have just completed this context, the engine may now be
-	 * idle and we want to re-enter powersaving.
-	 */
-	if (list_is_last(&rq->link, &ce->timeline->requests) &&
-	    i915_request_completed(rq))
-		intel_engine_add_retire(engine, ce->timeline);
+	if (INTEL_GEN(engine->i915) >= 10) {
+		/*
+		 * If we have just completed this context, the engine
+		 * may now be idle and we want to re-enter powersaving.
+		 */
+		if (list_is_last(&rq->link, &ce->timeline->requests) &&
+		    i915_request_completed(rq))
+			intel_engine_add_retire(engine, ce->timeline);
+	}
 
 	ccid >>= GEN11_SW_CTX_ID_SHIFT - 32;
 	ccid &= GEN12_MAX_CONTEXT_HW_ID;
-- 
2.40.1

